### PR TITLE
Follow up PR on image config controller feedback

### PR DESCRIPTION
**Pull request by m1kola**
### Which issue this PR addresses:

Followup PR for remaining feedback from #1766 (some of the comments are hidden by GitHub UI).

### Test plan for issue:

This PR itself is mostly about improving unit tests. Better coverage added for one case in `getCloudAwareRegistries`.

### Is there any documentation that needs to be updated for this PR?

No, just bug fixes and test improvements.

### CODE REVIEW

File: cmd/aro/operator.go [hunk 1]
```diff
@@ -188,7 +188,7 @@ func operator(ctx context.Context, log *logrus.Entry) error {
 			return fmt.Errorf("unable to create controller MachineSet: %v", err)
 		}
 		if err = (imageconfig.NewReconciler(
-			log.WithField("controller", controllers.ImageConfigControllerName),
+			controllers.ImageConfigControllerName),
```
Comment (bryanro92) [inline cmd/aro/operator.go:191]: doesn't compile, `controllers.ImageConfigControllerName),` leftover from the logrus instantiation.
```suggestion
```

File: pkg/operator/controllers/imageconfig/image_controller_test.go [hunk 2]
```diff
@@ -5,71 +5,49 @@ package imageconfig
...
 			name: "Feature Flag disabled, no action",
 			arocli: arofake.NewSimpleClientset(&arov1alpha1.Cluster{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: arov1alpha1.SingletonClusterName,
-				},
+				ObjectMeta: metav1.ObjectMeta{Name: arov1alpha1.SingletonClusterName},
 				Spec: arov1alpha1.ClusterSpec{
 					ACRDomain:     "arointsvc.azurecr.io",
-					AZEnvironment: "AzurePublicCloud",
+					AZEnvironment: azureclient.PublicCloud.Environment.Name,
```
Comment (bryanro92) [inline pkg/operator/controllers/imageconfig/image_controller_test.go:42]: :+1:

File: pkg/operator/controllers/imageconfig/image_controller_test.go [hunk 3]
```diff
@@ -205,35 +196,73 @@ func TestImageConfigReconciler(t *testing.T) {
...
+					Location:      "eastus",
+				},
+			},
+			wantErr: "cloud environment FakeCloud is not supported",
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			result, err := getCloudAwareRegistries(tt.instance)
+
+			if err != nil && err.Error() != tt.wantErr ||
```
Comment (bryanro92) [inline pkg/operator/controllers/imageconfig/image_controller_test.go:258]: makes sense :+1:, but I might've stared at this longer than most to understand each condition :sweat_smile: 

if no unexpected errors occurred unless the error is the error we want, or no errors occurred and we wanted an error, have the test return error.

**Review by bryanro92 — DISMISSED** [review]
small error to fix, but otherwise lgtm

**Comment by karanmagdani** [comment]
Apart from the leftover line already mentioned by Ross in operator.go , LGTM! 

I particularly like how you refactored the tests to separate out Public, Gov, and unsupported cloud. Much cleaner this way. 

Thank you @m1kola

**Review by karanmagdani — COMMENTED** [review]
only nit which is already mentioned by Ross, LGTM otherwise

File: pkg/operator/controllers/imageconfig/image_controller_test.go (hunk 3 above)
Comment (m1kola) [reply pkg/operator/controllers/imageconfig/image_controller_test.go:258]: This is the very common pattern we use across the codebase. But I agree - readability it not the best.

**Review by petrkotas — APPROVED** [review]
Thanks for this update. LGTM

**Pull request merged**